ops/ops/interface_aws/requires.py

```diff
@@ -130,14 +130,22 @@ def _joined(self, _):
             self.instance_id,
             self.region,
         )
-        self._to_publish["instance-id"] = self.instance_id
-        self._to_publish["region"] = self.region
+        self._request({"instance-id": self.instance_id, "region": self.region})
 
     @property
     def is_ready(self):
         completed = json.loads(self._received.get("completed", "{}"))
         response_hash = completed.get(self.instance_id)
-        return response_hash == self._expected_hash
+        ready = response_hash == self._expected_hash
+        if not response_hash:
+            log.warning("Remote end is yet to calculate a response")
+        elif not ready:
+            log.warning(
+                "Waiting for response_hash=%s to be self._expected_hash=%s",
+                response_hash,
+                self._expected_hash,
+            )
```

Comment:
Add some handy logging to figure out of the requires side has seen the correct hash yet.